### DISCUSSION TITLE
Logest snippet prefix to match (LSP Style snippets)

### DISCUSSION THREAD

**Discussion by hinell** [L3MON4D3/LuaSnip#890] [Q&A] [answered]
```json
{
    "vim log a": {
        "prefix": "log",
         ...
    }
    ,"vim log b": {
        "prefix": "api.log",
         ...
    }
    ,"vim log c": {
        "prefix": "vim.api.log",
         ...
    }
}
```

Hi.

Given the above VS Code / LSP Style snippets example, is there anyway I can easily configure `luasnip` to match the longest prefix on the line (and therefore expand that snippet) when `luasnip.complete()` is invoked?

Thanks!

@L3MON4D3 CC

**Comment by L3MON4D3** [comment, marked as answer]
Oh, you can do this is by setting separate priorities for each snippet, longest -> highest, shortest -> lowest
